From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Mon, 2 Feb 2026 19:11:03 +0000
Subject: [PATCH] Bug 2012313 - fix build with glibc-2.43 assignment discards
 'const' qualifier from pointer. r=nss-reviewers,keeler

Fixes:
    In file included from lowhash_vector.c:65:
    genload.c: In function 'loader_LoadLibInReferenceDir':
    genload.c:92:7: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
       92 |     c = strrchr(referencePath, PR_GetDirectorySeparator());
          |       ^

Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>

Differential Revision: https://phabricator.services.mozilla.com/D281507
---
 lib/freebl/genload.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/freebl/genload.c b/lib/freebl/genload.c
index 832deb58c470..e3231e3e527f 100644
--- a/lib/freebl/genload.c
+++ b/lib/freebl/genload.c
@@ -85,7 +85,7 @@ loader_LoadLibInReferenceDir(const char* referencePath, const char* name)
 {
     PRLibrary* dlh = NULL;
     char* fullName = NULL;
-    char* c;
+    const char* c;
     PRLibSpec libSpec;
 
     /* Remove the trailing filename from referencePath and add the new one */
